lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -3605,12 +3605,6 @@ void ByteCodeGenerator::EmitScopeList(ParseNode *pnode, ParseNode *breakOnBodySc
 
                 this->EmitOneFunction(pnode);
                 this->EndEmitFunction(pnode);
-
-                if (paramScope && !paramScope->GetCanMergeWithBodyScope())
```

Comment:
Nit: Similarly the pushing of bodyscope can also be moved to StartEmitFunction. 

---
In reply to: [61647086](https://github.com/Microsoft/ChakraCore/pull/892#discussion_r61647086) [](ancestors = 61647086)